### PR TITLE
Introduce 'modulesExcluded' property for ossIndexAudit

### LINKED ISSUE #109: [FEATURE] Introduce 'modulesExcluded' property for ossIndexAudit
* **What are you trying to do?**
Run an audit in a multi project situation and wanting to exclude the dependencies for a particular sub project. That sub project has a low priority in terms of maintenance and is only used within our own network where we control the network traffic.

* **What feature or behavior is this required for?**
With the new property we would not get false positives.

* **How could we solve this issue? (Not knowing is okay!)**
The nexusIQScan already supports this setting. Add a similar setting to ossIndexAudit. Not sure where this could best be implemented in the plugin. I'm happy to write a PR if you can point me in the right direction.

cc @bhamail / @DarthHater / @guillermo-varela / @shaikhu

### PR DESCRIPTION
**Pull request by TheoLassonder**
Similar to the property of the same name for nexusIQScan. Changes are as suggested by @guillermo-varela in issue #109.

Fixes #109 

cc @bhamail / @DarthHater / @guillermo-varela / @shaikhu

### CODE REVIEW

**Review by guillermo-varela — APPROVED** [review]
+1
Thanks for your contribution!

**Comment by guillermo-varela** [comment]
Hi @TheoLassonder,

Apparently there are some tests failures: https://circleci.com/gh/sonatype-nexus-community/scan-gradle-plugin/3087

Could you please take a look?

**Comment by TheoLassonder** [comment]
Hi @guillermo-varela, it looks like you approved one commit but not the second commit that fixes the test failure: [ece4718](https://github.com/sonatype-nexus-community/scan-gradle-plugin/pull/111/commits/ece4718130207f37e02801d441f5c8d480cd6b1a). Can you approve that change also?

Thanks.

**Review by guillermo-varela — APPROVED** [review]

**Comment by guillermo-varela** [comment]
Hi @TheoLassonder!

> it looks like you approved one commit but not the second commit that fixes the test failure: [ece4718](https://github.com/sonatype-nexus-community/scan-gradle-plugin/pull/111/commits/ece4718130207f37e02801d441f5c8d480cd6b1a). Can you approve that change also?

There are actually 3 commits in this PR and the one you mention addresses some errors related to the first commit.

However, the test failure is being caused by the third commit, fd1746863681929050bad0dc04d1b63995bd0954, as that one introduced the 2 new test methods failing at CI: `testAudit_verifyModulesIncludedExist` and `testAudit_verifyModulesExcludedAppliedAfterModulesIncluded`:

https://github.com/sonatype-nexus-community/scan-gradle-plugin/pull/111/commits/fd1746863681929050bad0dc04d1b63995bd0954#diff-1f7c62b0958793275fa106f008b1b47f2f49767537d893e7b61780fa7a4fe383R94-R117

https://app.circleci.com/pipelines/github/sonatype-nexus-community/scan-gradle-plugin/557/workflows/7c116021-139e-4e86-b90b-c8e55d22a8db/jobs/3088/tests

File: src/main/java/org/sonatype/gradle/plugins/scan/ossindex/OssIndexPluginExtension.java [hunk 1]
```diff
@@ -44,6 +44,10 @@
 
   private boolean allConfigurations;
 
+  private Set<String> modulesIncluded;
+
+  private Set<String> modulesExcluded;
```
Comment (guillermo-varela) [inline src/main/java/org/sonatype/gradle/plugins/scan/ossindex/OssIndexPluginExtension.java:49]: Could you please add this 2 properties at [OssIndexAuditTask.java](https://github.com/sonatype-nexus-community/scan-gradle-plugin/blob/fd1746863681929050bad0dc04d1b63995bd0954/src/main/java/org/sonatype/gradle/plugins/scan/ossindex/OssIndexAuditTask.java#L248-L250) with the `@Input` annotation at the bottom?

Comment (TheoLassonder) [reply src/main/java/org/sonatype/gradle/plugins/scan/ossindex/OssIndexPluginExtension.java:49]: Sure, thanks for pointing that out.

**Comment by TheoLassonder** [comment]
@[guillermo-varela](https://github.com/guillermo-varela) Apologies for the failing tests! Fixed now

File: src/test/java/org/sonatype/gradle/plugins/scan/ossindex/OssIndexAuditTaskTest.java [hunk 2]
```diff
@@ -47,6 +48,7 @@
 import static org.gradle.api.plugins.JavaPlugin.COMPILE_CLASSPATH_CONFIGURATION_NAME;
 import static org.mockito.ArgumentMatchers.anyList;
 import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.atMost;
```
Comment (guillermo-varela) [inline src/test/java/org/sonatype/gradle/plugins/scan/ossindex/OssIndexAuditTaskTest.java:51]: It looks like this import is not used, if so please remove it

**Comment by TheoLassonder** [comment]
So everything is OK now?

**Comment by guillermo-varela** [comment]
> So everything is OK now?

Yes it is @TheoLassonder!

We greatly appreciate your patience and commitment to get this features done :)

**Pull request merged**